quizcon/main/templatetags/quiz_tools.py

```diff
@@ -141,51 +141,43 @@ def percentage_choice(x, question):
 
 
 @register.simple_tag
-def right_answers_per_quiz(id):
+def questions_most_correct(id):
     quiz = Quiz.objects.get(pk=id)
-    num_of_correct = 0
-    question_most_correct = []
-    questions = {}
+    num_of_correct = {}
     for question in quiz.question_set.all():
-        if num_of_correct < total_right_answers(question):
-            num_of_correct = total_right_answers(question)
+        num_of_correct.update({question: total_right_answers(question)})
+    max_num_correct = max(num_of_correct.values())
 
-        questions.update({total_right_answers(question): question})
-    for key in questions:
-        if key == num_of_correct:
-            question_most_correct.append(questions[key])
-    return question_most_correct
+    if max_num_correct == 0:
+        return "No questions answered correctly."
```

Comment:
Suggestion for all three of the functions. Maybe return an empty array if the result == 0? Then, in the template, you can check the length of the array and print out the error message. This approach keeps all the messaging in the template.